src/main/scala/com/fulcrumgenomics/bam/EstimatePoolingFractions.scala

```diff
@@ -0,0 +1,242 @@
+    val regression = new OLSMultipleLinearRegression
+    regression.setNoIntercept(true) // Intercept should be at 0!
+    regression.newSampleData(
+      coveredLoci.map(_.observedFraction.getOrElse(unreachable("observed fraction must be defined"))),
+      coveredLoci.map(_.expectedSampleFractions)
+    )
+
+    val regressionParams = regression.estimateRegressionParameters()
+    val total            = regressionParams.sum
+    val fractions        = regressionParams.map(_ / total)
+    val stderrs          = regression.estimateRegressionParametersStandardErrors().map(_ / total)
+    logger.info(s"R^2 = ${regression.calculateRSquared()}")
+    logger.info(s"Sum of regression parameters = ${total}")
+
+    val metrics = sampleNames.toSeq.zipWithIndex.map { case (sample, index) =>
+      val sites      = coveredLoci.count(l => l.expectedSampleFractions(index) > 0)
+      val singletons = coveredLoci.count(l => l.expectedSampleFractions(index) > 0 && l.expectedSampleFractions.sum == l.expectedSampleFractions(index))
+      new PoolingFractionMetric(
+        sample             = sample,
+        variant_sites      = sites,
+        singletons         = singletons,
+        estimated_fraction = fractions(index),
+        standard_error     = stderrs(index),
+        ci99_low           = max(0, fractions(index) - stderrs(index)*Ci99Width),
+        ci99_high          = min(1, fractions(index) + stderrs(index)*Ci99Width))
+    }
+
+    Metric.write(metrics, path=output)
+
+    if (regression.estimateRegressionParameters().exists(_ < 0)) {
+      logger.error("#################################################################################")
+      logger.error("# One or more samples is estimated to have fraction < 0. This is likely due to  #")
+      logger.error("# insufficient coverage at some loci and/or too few SNPs being used.            #")
+      logger.error("# Please try again with a higher min-coverage-per-sample.                       #")
+      logger.error("#################################################################################")
+      fail(1)
+    }
+  }
+
+  /** Verify a provided sample list, or if none provided retrieve the set of samples from the VCF. */
+  private def pickSamplesToUse(vcfReader: VCFFileReader): Array[String] = {
+    if (samples.nonEmpty) {
+      val samplesInVcf = vcfReader.getFileHeader.getSampleNamesInOrder.toSet
+      val missingSamples = samples.filterNot(samplesInVcf.contains)
+      if (missingSamples.nonEmpty) fail(s"Samples not present in VCF: ${missingSamples.mkString(", ")}")
+      else samples.toArray.sorted
+    }
+    else {
+      vcfReader.getFileHeader.getSampleNamesInOrder.toSeq.toArray.sorted // toSeq.toArray is necessary cos util.ArrayList.toArray() exists
+    }
+  }
+
+  /** Loads up and merges all the interval lists provided. Returns None if no intervals were specified. */
+  private def loadIntervals: Option[IntervalList] = {
```

Comment:
As a style comment, methods without parameters in my mind should look like `val`s, and so should be named as such.  What do you think?